### PR TITLE
fix(widget)!: List should not ignore empty string items

### PR DESCRIPTION
**Pull request by orhun**
> Upstream: [#688](https://github.com/fdehau/tui-rs/pull/688)

## Description
Fixes issue [#680](https://github.com/fdehau/tui-rs/issues/680#issue-1462518032). Handles the case where a list item is created with an empty string, which is not split by the `lines` iterator.

## Testing guidelines
New test can be ran with `cargo test`

## Checklist

* [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
* [x] I have added relevant tests.
* [x] I have documented all new additions.

### CODE REVIEW

File: src/text.rs [hunk 1]
```diff
@@ -292,10 +292,16 @@ impl<'a> Text<'a> {
...
+        let lines: Vec<_> = match content.into() {
+            Cow::Borrowed(s) => s.lines().map(Spans::from).collect(),
+            Cow::Owned(s) => s.lines().map(|l| Spans::from(l.to_owned())).collect(),
+        };
+
         Text {
-            lines: match content.into() {
-                Cow::Borrowed(s) => s.lines().map(Spans::from).collect(),
-                Cow::Owned(s) => s.lines().map(|l| Spans::from(l.to_owned())).collect(),
+            lines: if lines.len() == 0 {
```
Comment (sayanarijit) [inline src/text.rs:301]:
```suggestion
            lines: if lines.is_empty() {
```

**Review by sayanarijit — CHANGES_REQUESTED** [review]

**Review by mindoodoo — APPROVED** [review]

**Comment by rhysd** [comment]
This branch needs to be rebased on the latest main branch so that newer version of Rust compiler will be used and it will fix the issue on compiling `cargo-make` crate.

**Comment by orhun** [comment]
Rebased.

**Review by rhysd — APPROVED** [review]

**Review by sayanarijit — CHANGES_REQUESTED** [review]
Please hold this, I think I missed to see this, it can be improved. is_empty() should be done in the input string.

**Comment by sayanarijit** [comment]
Updated.

**Comment by sayanarijit** [comment]
But this looks like a breaking change. Some users might be using this bug/feature as a condition, So, let's wait for a major release?

**Comment by sayanarijit** [comment]
Ah, good point.

**Review by sayanarijit — APPROVED** [review]

**Review by mindoodoo — APPROVED** [review]

**Pull request merged**